### PR TITLE
Add Trino 407 release notes

### LINKED ISSUE #15854: Release notes for 407

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 407 release. Fixes #15854 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 25 Jan 2023

@ebyhr #15806 ✅ rn ✅ docs
@ebyhr #15852 ✅ rn ✅ docs
@ebyhr #15851 ✅ rn ✅ docs

## 26 Jan 2023

@findepi #15350 ✅ rn ✅ docs
@raunaqmorarka #15713 ✅ rn ✅ docs
@ebyhr #15807 ✅ rn ✅ docs
@ebyhr #15857 ✅ rn ✅ docs
@findepi #15841 ✅ rn ✅ docs
@findepi #15844 ✅ rn ✅ docs
@sopel39 #15838 ✅ rn ✅ docs
@electrum #15703 ✅ rn ✅ docs
@findepi #15843 ✅ rn ✅ docs
@findepi #15825 ✅ rn ✅ docs
@findepi #15863 ✅ rn ✅ docs
@ebyhr #15855 ✅ rn ✅ docs
@ebyhr #15866 ✅ rn ✅ docs

## 27 Jan 2023

@findepi #15849 ✅ rn ✅ docs
@findepi #15862 ✅ rn ✅ docs
@findepi #15861 ✅ rn ✅ docs
@findepi #15744 ✅ rn ✅ docs
@findepi #15860 ✅ rn ✅ docs
@sopel39 #15875 ✅ rn ✅ docs
@martint #15085 ✅ rn ✅ docs
@ebyhr #15876 ✅ rn ✅ docs

## 28 Jan 2023

@martint #15886 ✅ rn ✅ docs
@electrum #15833 ✅ rn ✅ docs

## 30 Jan 2023

@findepi #15879 ✅ rn ✅ docs
@raunaqmorarka #15850 ✅ rn ✅ docs
@kasiafi #14566 ✅ rn ✅ docs
@kokosing #15871 ✅ rn ✅ docs
@sopel39 #15864 ✅ rn ✅ docs
@arhimondr #15721 ✅ rn ❌ docs
@arhimondr #15901 ✅ rn ✅ docs
@ebyhr #15889 ✅ rn ✅ docs
@ebyhr #15902 ✅ rn ✅ docs
@raunaqmorarka #15858 ✅ rn ✅ docs

## 31 Jan 2023

@Praveen2112 #15823 ✅ rn ✅ docs
@ebyhr #15906 ✅ rn ✅ docs
@ebyhr #15905 ✅ rn ✅ docs
@findepi #15896 ✅ rn ✅ docs
@sopel39 #15811 ✅ rn ✅ docs
@sopel39 #15899 ✅ rn ✅ docs
@ebyhr #15436 ✅ rn ✅ docs
@ebyhr #15094 ✅ rn ❌ docs
@raunaqmorarka #15897 ✅ rn ✅ docs
@ebyhr #15914 ✅ rn ✅ docs

## 1 Feb 2023

@findepi #15909 ✅ rn ✅ docs
@findepi #15912 ✅ rn ✅ docs
@findepi #15908 ✅ rn ✅ docs
@sopel39 #15579 ✅ rn ✅ docs
@sopel39 #15924 ✅ rn ✅ docs
@dain #15775 ✅ rn ✅ docs
@losipiuk #15870 ✅ rn ✅ docs
@ebyhr #15922 ✅ rn ✅ docs

## 2 Feb 2023

@findepi #15929 ✅ rn ✅ docs
@raunaqmorarka #15923 ✅ rn ✅ docs
@findepi #15939 ✅ rn ✅ docs
@findepi #15936 ✅ rn ✅ docs
@raunaqmorarka #15942 ✅ rn ✅ docs
@ebyhr #15958 ✅ rn ✅ docs

## 3 Feb 2023

@sopel39 #15489 ✅ rn ❌ docs
@sopel39 #15948 ✅ rn ✅ docs
@ebyhr #15856 ✅ rn ✅ docs
@findepi #15957 ✅ rn ✅ docs
@ebyhr #15959 ✅ rn ✅ docs
@ebyhr #15940 ✅ rn ✅ docs
@sopel39 #15949 ✅ rn ✅ docs
@sopel39 #15839 ✅ rn ✅ docs

## 4 Feb 2023

@ebyhr #15893 ✅ rn ✅ docs

## 5 Feb 2023

@kokosing #15781 ✅ rn ✅ docs
@ebyhr #15747 ✅ rn ✅ docs

## 6 Feb 2023

@raunaqmorarka #15954 ✅ rn ✅ docs
@ebyhr #15396 ✅ rn ✅ docs
@sopel39 #15983 ✅ rn ✅ docs
@findepi #15985 ✅ rn ✅ docs
@electrum #15702 ✅ rn ✅ docs
@electrum #14020 ✅ rn ✅ docs
@electrum #15765 ✅ rn ✅ docs
@findepi #15638 ✅ rn ✅ docs
@findepi #15969 ✅ rn ✅ docs
@raunaqmorarka #13814 ✅ rn ✅ docs

## 7 Feb 2023

@losipiuk #15947 ✅ rn ✅ docs
@findepi #15878 ✅ rn ✅ docs
@raunaqmorarka #16001 ✅ rn ✅ docs
@raunaqmorarka #15816 ✅ rn ✅ docs
@kasiafi #15989 ✅ rn ✅ docs
@findepi #15981 ✅ rn ✅ docs
@ebyhr #15996 ✅ rn ✅ docs
@Praveen2112 #15935 ✅ rn ✅ docs

## 8 Feb 2023

@findepi #15984 ✅ rn ✅ docs
@findepi #15453 ✅ rn ✅ docs
@findepi #14373 ✅ rn ✅ docs
@martint #16000 ✅ rn ✅ docs
@kasiafi #15994 ✅ rn ✅ docs
@arhimondr #15933 ✅ rn ✅ docs
@findepi #16017 ✅ rn ✅ docs

## 9 Feb 2023

@kasiafi #16010 ✅ rn ✅ docs
@hashhar #15724 ✅ rn ✅ docs
@findepi #15797 ✅ rn ✅ docs
@kokosing #15941 ✅ rn ✅ docs
@ebyhr #16044 ✅ rn ✅ docs
@electrum #16049 ✅ rn ✅ docs
@kasiafi #16014 ✅ rn ✅ docs
@electrum #15577 ✅ rn ✅ docs
@findepi #16043 ✅ rn ✅ docs
@electrum #14402 ✅ rn ✅ docs
@arhimondr #16031 ✅ rn ✅ docs
@ebyhr #16028 ✅ rn ✅ docs
@raunaqmorarka #15883 ✅ rn ✅ docs
@raunaqmorarka #15962 ✅ rn ✅ docs
@Praveen2112 #15960 ✅ rn ✅ docs

## 10 Feb 2023

@ebyhr #16040 ✅ rn ✅ docs
@ebyhr #16069 ✅ rn ✅ docs

## 11 Feb 2023

@ebyhr #16032 ✅ rn ✅ docs
@kasiafi #16058 ✅ rn ✅ docs

## 12 Feb 2023

@ebyhr #16047 ✅ rn ✅ docs
@raunaqmorarka #15986 ✅ rn ✅ docs
@ebyhr #16062 ✅ rn ✅ docs
@ebyhr #15788 ✅ rn ✅ docs

## 13 Feb 2023

@arhimondr #15917 ✅ rn ✅ docs
@ebyhr #16097 ✅ rn ✅ docs

## 14 Feb 2023

@kokosing #16060 ✅ rn ✅ docs
@Praveen2112 #15898 ✅ rn ✅ docs
@ebyhr #15956 ✅ rn ✅ docs
@arhimondr #16111 ✅ rn ✅ docs
@ebyhr #15874 ✅ rn ✅ docs
@ebyhr #16108 ✅ rn ✅ docs
@ebyhr #16117 ✅ rn ✅ docs
@ebyhr #15556 ✅ rn ✅ docs
@ebyhr #16118 ✅ rn ✅ docs
@raunaqmorarka #16052 ✅ rn ✅ docs

## 15 Feb 2023

@ebyhr #16007 ✅ rn ✅ docs
@ebyhr #15973 ✅ rn ✅ docs
@raunaqmorarka #16110 ✅ rn ✅ docs
@ebyhr #16055 ✅ rn ✅ docs
@kokosing #16121 ✅ rn ✅ docs
@kokosing #16086 ✅ rn ✅ docs
@arhimondr #16113 ✅ rn ✅ docs
@electrum #16133 ✅ rn ❌ docs
@ebyhr #14869 ✅ rn ✅ docs

## 16 Feb 2023

@martint #16134 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:5]:
```suggestion
* Add support for adaptive task request sizing. It can be enabled with the
```

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:11]: Is "always" needed here?

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:15]: maybe use "three"

**Review by mosabua — APPROVED** [review]

**Comment by colebow** [comment]
Looks like we're blocked by #15931

**Comment by arhimondr** [comment]
Here is the fix: https://github.com/trinodb/trino/pull/15933

File: docs/src/main/sphinx/release/release-407.md
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-407.md:29]: I think #15811 reduces query latency in Delta Lake connector as well. 
cc: @sopel39 @findepi

Comment (findepi) [reply docs/src/main/sphinx/release/release-407.md:29]: I don't think it does. Delta connector doesn't read stats from the metastore.

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:1]: 13

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:20]: how many nodes ??

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:22]: Is the docs updated?

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:23]: I am not a friend of using "now" .. especially not twice..

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:23]: Could this even be a separate release notes entry .. but with the same PR number?

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:32]: fails, but

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:90]: is it really called a Kerberos ticket ?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-407.md
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:22]: No. ☹️

File: docs/src/main/sphinx/release/release-407.md
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:32]: "reports normal completion" isn't an independent clause, so this comma would be incorrect

File: docs/src/main/sphinx/release/release-407.md
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:90]: [Yep!](https://web.mit.edu/kerberos/krb5-latest/doc/user/tkt_mgmt.html)

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:91]: Wow.. crazy

File: docs/src/main/sphinx/release/release-407.md
Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md]: Sign. .. can you send a PR ?

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:12]: Has the default changed? What's the user visible impact? If this is not visible, remove the entry. 

Also, why are these "optimizer.xxx" options? The seem to be related to the runtime, in which case they should be "query.xxxx". cc @gaurav8297 @sopel39

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:28]: @findepi , why did this change? We discussed this being incompatible with the SQL spec, so it shouldn't have gone in.

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:30]: UNKNOWN is not a type. If anything, this should say "when processing null values". 

However, the entry should be talking about the user-visible impact. What does "Fix handling" mean? Was there a failure? A performance problem? When does it manifest? What kinds of queries trigger it?

cc @lukasz-stec @sopel39

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:44]: Remove "non-optimized". Just say "the legacy Parquet writer".

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:55]: The proper name is "Apache Impala".

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:69]: What does "adding xxx property to the JDBC catalog"? What's the functionality being added? Should this be "Add support for specifying a user and password when connecting to the JDBC catalog"?

cc @ebyhr

File: docs/src/main/sphinx/release/release-407.md
Comment (lukasz-stec) [reply docs/src/main/sphinx/release/release-407.md:30]: AFAIK, there are no real queries that could trigger this in the current code. I would drop the release note.

File: docs/src/main/sphinx/release/release-407.md
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:26]: Can we revert to unblock the release, then discuss this later?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:26]: I agree with reverting for now so we can get a release out.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:26]: cc @jhlodin

**Review by mosabua — APPROVED** [review]
All looks good. A bunch of things are probably missing documentation updates though.

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:23]: ORC

File: docs/src/main/sphinx/release/release-407.md
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:30]: Why would a query fail and report normal completion? Or is the bug that it fails and reports normal completion?

If so, I'd rephrase as: "Fix potential incorrect results due to query reporting normal completion instead of failing"

**Pull request merged**
